CompressedFluids/prototypes/fluid/compressor-fluid-updates.lua

```diff
@@ -1,53 +1,60 @@
-          group = "compressed-fluids",
-          order = data.raw["item-group"][data.raw["item-subgroup"][fluidPrototype.subgroup].group].order
-        }
-      }
-    end
-    fluidPrototype.order = data.raw["item-subgroup"][fluidPrototype.subgroup].order.."["..fluidPrototype.subgroup.."]-"..fluidPrototype.order
-    fluidPrototype.subgroup = "compressed-fluids["..data.raw["item-subgroup"][fluidPrototype.subgroup].group.."]"
-
-    fluidPrototype.flags = fluidPrototype.flags or {}
-    table.insert(fluidPrototype.flags, "hidden")
-
-    data:extend{fluidPrototype}
-  end
-end
+CF_compressedFluidNames = {}
+
+for fluidName,fluidPrototype in pairs(util.table.deepcopy(data.raw.fluid)) do
+  -- If the fluid is barrelable, we create the HP fluid
+  if (fluidPrototype.auto_barrel == nil) or (fluidPrototype.auto_barrel == "true") or (fluidPrototype.auto_barrel == true) then
+    fluidPrototype.auto_barrel = false
+
+    -- name
+    fluidPrototype.name = "high-pressure-"..fluidName
+    CF_compressedFluidNames[fluidName] = fluidPrototype.name
+    
+    fluidPrototype.localised_name = {"fluid-name.compressed-fluid", fluidPrototype.localised_name or {"fluid-name."..fluidName}}
+
+    -- icon
+    if fluidPrototype.icon and (not fluidPrototype.icons) then
+      fluidPrototype.icons = {{icon = fluidPrototype.icon}}
+    end
+    fluidPrototype.icon = nil
+    if fluidPrototype.icon_size and fluidPrototype.icons and fluidPrototype.icons[1] and (not fluidPrototype.icons.icon_size) then
+      for k,_ in pairs(fluidPrototype.icons) do
+        fluidPrototype.icons[k].icon_size = fluidPrototype.icon_size
+      end
+    end
+    fluidPrototype.icon_size = nil
+
+    table.insert(fluidPrototype.icons,  {
+      icon = "__CompressedFluids__/graphics/icons/overlay-HP.png",
+      icon_size = 32,
+      tint = {r=0,g=1,b=0}
+    })
+
+    -- Fluid properties
+    fluidPrototype.fuel_value = nil
+    fluidPrototype.max_temperature = fluidPrototype.default_temperature
+
+    -- Move compressed fluids to a new group and create subgroup for it
+    fluidPrototype.subgroup = fluidPrototype.subgroup or "fluid-recipes" -- make sure it has a subgroup
+    fluidPrototype.order = fluidPrototype.order or "zzz"                 -- make sure it has an order string
+    -- If our subgroup is nil, insert a raw table
+    local itemSubgroup = data.raw["item-subgroup"][fluidPrototype.subgroup] or {}
+    if not data.raw["item-subgroup"]["compressed-fluids["..itemSubgroup.group or "".."]"] then
+      data:extend{
+        {
+          type = "item-subgroup",
+          name = "compressed-fluids["..itemSubgroup.group.."]",
+          group = "compressed-fluids",
+          order = data.raw["item-group"][itemSubgroup.group].order
+        }
+      }
+    end
+
+    fluidPrototype.order = itemSubgroup.order or "".."["..fluidPrototype.subgroup.."]-"..fluidPrototype.order
+    fluidPrototype.subgroup = "compressed-fluids["..itemSubgroup.group or "".."]"

```

Comment:
I'm not 100% sure what the root cause for the issue is, so this is a blind fix. It does work in practice but I'd welcome any cleanup or suggestions.

Reply:
In the line below, `itemSubgroup.group` can still be nil, assuming your statement above it turns out to be `{}`.
https://github.com/LovelySanta/FactorioMod-OtherSmallMods/blob/dbdf4ef1754fad2ef92374489adc22c6ad722f9d/CompressedFluids/prototypes/fluid/compressor-fluid-updates.lua#L45

I don't see how this can become `{}` in the first place, meaning the whole PR has no use? I've contacted you already on discord, please provide me a screenshot of the error and maybe which mod that is causing this issue alongside CompressedFluids?